packages/api/src/platforms/vtex/resolvers/validateSession.ts

```diff
@@ -29,6 +29,9 @@ export const validateSession = async (
 
   const profile = sessionData?.namespaces.profile ?? null
   const store = sessionData?.namespaces.store ?? null
+  const region = regionData?.[0]
+  // Set seller only if it's inside a region
+  const seller = region?.sellers.find(seller => channel.seller === seller.id)
```

Comment:
If I understood well, in the first `validateSession` request with `regionData`, the `channel.seller` would be `undefined` and this `.find` would return undefined. 

Should we include a fallback `?? region?.sellers?.[0]` ?

Reply:
I don't think so, @eduardoformiga. In this case, attributing the seller variable to the first seller would force the selected seller to always be the first one. If it's undefined, the API will default to what is now the default behavior: users can see offers from any sellers.

Reply:
That's true. 
I think with this logic the problem remains, as we only set the `channel.seller` after this line. This `.find()` method will always return `undefined` from now on, wouldn't it?

Reply:
The `.find()` method is called just once when this line runs.  I don't see any problem with the logic. Could you elaborate?

Reply:
That's right, @icazevedo. I didn't see the [mention](https://github.com/vtex-sites/casinofr.store/blob/f6a1bfc653d209caed33fd3e41f6b41326d9ad74/src/components/regionalization/contexts/RegionalizationContext.tsx#L123) of the casino's code in the description and was thinking that we just set the seller here on the API code. Now, I can see that we set it there from the store code, and also I tested it using and looks good. I'll resolve this conversation ;)